openmetadata-service/src/main/java/org/openmetadata/service/migration/utils/v140/MigrationUtil.java

```diff
@@ -66,62 +66,68 @@ public static void migrateTablePartition(Handle handle, CollectionDAO collection
   }
 
   private static void handleTablePartitionMigration(String jsonRow, CollectionDAO collectionDAO) {
-    JsonObject jsonObj = JsonUtils.readJson(jsonRow).asJsonObject();
-
-    // We need to pop the tablePartition from the json before serializing it
-    JsonObject tablePartition = jsonObj.getJsonObject("tablePartition");
+    try {
+      JsonObject jsonObj = JsonUtils.readJson(jsonRow).asJsonObject();
 
-    // Remove the tablePartition from the json. We need to convert it to a map to remove the key
-    // as JsonObject is immutable
-    HashMap<String, Object> jsonMap = JsonUtils.readValue(jsonObj.toString(), HashMap.class);
-    jsonMap.remove("tablePartition");
+      // We need to pop the tablePartition from the json before serializing it
+      JsonObject tablePartition = jsonObj.getJsonObject("tablePartition");
 
-    jsonObj = JsonUtils.readJson(JsonUtils.pojoToJson(jsonMap)).asJsonObject();
+      // Remove the tablePartition from the json. We need to convert it to a map to remove the key
+      // as JsonObject is immutable
+      HashMap<String, Object> jsonMap = JsonUtils.readValue(jsonObj.toString(), HashMap.class);
+      jsonMap.remove("tablePartition");
 
-    Table table = JsonUtils.readValue(jsonObj.toString(), Table.class);
+      jsonObj = JsonUtils.readJson(JsonUtils.pojoToJson(jsonMap)).asJsonObject();
 
-    JsonArray partitionColumns = tablePartition.getJsonArray("columns");
-    if (tablePartition.isEmpty()) {
-      LOG.info("Table {} does not have partition details", table.getId());
-      return;
-    }
+      Table table = JsonUtils.readValue(jsonObj.toString(), Table.class);
 
-    List<PartitionColumnDetails> partitionColumnDetails = new ArrayList<PartitionColumnDetails>();
-
-    if ((partitionColumns == null || partitionColumns.isEmpty())
-        && table.getServiceType() == CreateDatabaseService.DatabaseServiceType.BigQuery) {
-      // BigQuery tables have pseudo columns for partitioning that were not being set in the
-      // partitionColumns entity
-      String interval = tablePartition.getString("interval");
-      if (interval != null) {
-        JsonArrayBuilder jsonArrayBuilder = Json.createArrayBuilder();
-        switch (interval) {
-          case "HOUR" -> partitionColumns = jsonArrayBuilder.add("_PARTITIONTIME").build();
-          case "DAY" -> partitionColumns = jsonArrayBuilder.add("_PARTITIONDATE").build();
+      if (tablePartition.isEmpty()) {
+        LOG.info("Table {} does not have partition details", table.getId());
+        return;
```

Comment:
rule of thumb: there should be only one return for a method in java. Use the inverse condition here.